helm-charts/azure-api-management-gateway/values.yaml

```diff
@@ -17,8 +17,15 @@ gateway:
   configuration:
     uri: 
   auth:
+    # When an existing secret is configured, the value specified here will not be used
     key: 
 
+secret:
+  useExisting:
+    enabled: false
+    name: 
+
```

Comment:
Let's keep it flat and use `secret.name` when `secret.create: true`
```suggestion
secret:
  create: true
  name: 

```

Reply:
Using this syntax, wouldn't it be confusing to only use the name when using an existing secret.
Do I use secret.name value in the secret created by the helm chart as well?
With secret.name defaulting to "azure-api-management-gateway" in values.yaml?

Reply:
Fair point, what about this:
```suggestion
secret:
  createSecret: true
  secretName: ""

```

That way it's more clear and flat, thoughts?

Reply:
I'm happy to jump in and help on this reshuffling though, if you want.

Reply:
The suggested change should be implemented